Dockerfile.Ubuntu1604

```diff
@@ -14,38 +14,44 @@ RUN apt-get -qq update && apt-get -qq install \
     python-numpy \
     python-dev \
     wget \
-    bzip2
+    bzip2 && \
+    rm -rf /var/lib/apt/lists/*
 
 # Installing boost from source
+WORKDIR /boost-build
 RUN wget -nv 'https://dl.bintray.com/boostorg/release/1.65.1/source/boost_1_65_1.tar.bz2' -O - | tar xj && \
     cd boost_1_65_1 && \
     ./bootstrap.sh --with-libraries=log,thread,system,filesystem,program_options,test --prefix=/usr/local && \
     ./b2 -j$(nproc) install && \
+    rm -rf /boost-build && \
     ldconfig
 
 # Rebuild image if force_rebuild after that command
 ARG CACHEBUST
 
 # Setting some environment variables for installing preCICE
-ENV CPLUS_INCLUDE_PATH="$CPLUS_INCLUDE_PATH:/usr/include/eigen3"
-ENV CPATH="/usr/include/eigen3:${CPATH}"
-ENV PETSC_DIR="/usr/lib/petscdir/3.6.2/"
-ENV PETSC_ARCH="x86_64-linux-gnu-real"
+ENV CPLUS_INCLUDE_PATH="$CPLUS_INCLUDE_PATH:/usr/include/eigen3" \
+    CPATH="/usr/include/eigen3:${CPATH}" \
+    PETSC_DIR="/usr/lib/petscdir/3.6.2/" \
+    PETSC_ARCH="x86_64-linux-gnu-real"
 
 # Building preCICE
 ARG branch=develop
-RUN git clone --branch $branch https://github.com/precice/precice.git
-WORKDIR precice
+RUN git clone --branch $branch https://github.com/precice/precice.git /precice
+WORKDIR /precice
 # Some parameters for the build, you can set them in the build command e.g.
 # sudo docker build Dockerfile.precice --build-arg petsc_para=yes --build-arg mpi_para=yes .
 # this will result in
 # scons petsc=yes mpi=yes python=no compiler="mpicxx" -j2
 ARG petsc_para=no
 ARG mpi_para=yes
 ARG python_para=no
-RUN scons petsc=$petsc_para mpi=$mpi_para python=$python_para -j$(nproc)
+RUN scons petsc=$petsc_para mpi=$mpi_para python=$python_para -j$(nproc) && \
+    find . -type f \( -name "*.o" -or -name "*.os" \) -exec rm {} +
```

Comment:
Please add a comment, why this is necessary. Don't think it's obvious that's just for reducing image size.

Reply:
True, alternatively we could wipe the entire directory part from the library and executables.

Reply:
I added a comment regarding the clean-up.
Feel free to resolve the conversation if you are ok with it.